proxy/pkg/zdmproxy/clienthandler.go

```diff
@@ -1403,6 +1413,61 @@ func (ch *ClientHandler) forwardRequest(request *frame.RawFrame, customResponseC
 	return nil
 }
 
+func (ch *ClientHandler) modifyRequestIfNeeded(request *frame.RawFrame, currentKeyspace string) (*frameDecodeContext, []*statementReplacedTerms, error) {
+	context := NewFrameDecodeContext(request)
+	decodedFrame, err := context.GetOrDecodeFrame()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var statementsQueryData []*statementQueryData = nil
+	var replacedTerms = []*statementReplacedTerms{}
+	var modifiedFrame, modifiedQuery bool
+
+	// replace runtime-evaluated CQL functions in queries
+	if ch.conf.ReplaceCqlFunctions {
+		decodedFrame, statementsQueryData, err = context.GetOrDecodeAndInspect(currentKeyspace, ch.timeUuidGenerator)
+		if err != nil {
+			if !errors.Is(err, NotInspectableErr) {
+				return nil, nil, fmt.Errorf("could not check whether query needs replacement for a '%v' request: %w",
+					decodedFrame.Header.OpCode.String(), err)
+			}
```

Comment:
missing something here to handle the `else` scenario where `err` is still `nil` and therefore `decodedFrame` is `nil`